### PR TITLE
Add warning notice in checkout when plan includes backup already

### PR DESCRIPTION
**Pull request by monsieur-z**
#### Changes proposed in this Pull Request

Customers who purchased a Backup product (monthly or real time) can have a duplicated backup feature by purchasing a Jetpack plan that includes it as well. This PR adds a warning for such cases in the checkout screen.

#### Testing instructions

1. No Backup
    * Select a Jetpack site that has no Backup
    * Go to _Plan > _Plans_
    * Select any of the 3 plans
    * Once in the checkout page, verify that you **don't** see any notice and that there's no error in the console
    * Repeat for the other plans
2. Backup Daily
    * Select a Jetpack site that has Backup Daily
    * Go to _Plan > _Plans_
    * Select any of the 3 plans
    * Once in the checkout page, verify that you **do** see a notice and that there's no error in the console
    * Both links in the notice should redirect you to the Backup Daily purchase page
    * Repeat for the other plans
3. Backup Real-Time
    * Select a Jetpack site that has Backup Real-Time
    * Go to _Plan > _Plans_
    * Select any of the 3 plans
    * Once in the checkout page, verify that you **do** see a notice **only for the Business plan**
    * Both links in the notice should redirect you to the Backup Real-Time purchase page
    * Repeat for the other plans

#### Screenshots

**Mockups**
Screen Shot 2020-07-16 at 3 32 29 PM

**Implementation**
Screen Shot 2020-07-16 at 3 33 03 PM

### CODE REVIEW

File: client/my-sites/checkout/checkout/duplicate-product-notice-content.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+							product: product.productName,
+						},
+						components: {
+							link: <a href={ subscriptionUrl } />,
+						},
+					}
+				) }
+			</p>
+			<a className="checkout__duplicate-notice-link" href={ subscriptionUrl }>
+				{ translate( 'Manage Subscription' ) }
```
Comment (a8ci18n) [inline client/my-sites/checkout/checkout/duplicate-product-notice-content.tsx:47]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **33** times:
`translate( 'Manage subscription' )` *ES Score: 14*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/plan-backup-warning-in-checkout)

File: client/my-sites/checkout/checkout/style.scss [hunk 2]
```diff
@@ -468,28 +468,29 @@
...
-						 max-height: 21px;
-					 }
-				 }
-
-				 .checkout__pan-number-popover svg {
-					 max-height: 16px;
-				 }
-			 }
-		 }
+		.checkout__payment-box-section {
```
Comment (monsieur-z) [inline client/my-sites/checkout/checkout/style.scss:471]: Just some reformatting.

File: client/my-sites/checkout/checkout/secure-payment-form.jsx [hunk 3]
```diff
@@ -307,45 +308,19 @@ export class SecurePaymentForm extends Component {
 				products={ this.props.products }
 				selectedSite={ this.props.selectedSite }
 				transactionStep={ this.props.transaction.step }
+				infoMessage={ this.props.infoMessage }
 			/>
 		);
 	}
 
-	renderCreditCardPaymentBox() {
```
Comment (monsieur-z) [inline client/my-sites/checkout/checkout/secure-payment-form.jsx:314]: This method is never used.

**Review by wp-desktop — DISMISSED** [review]
WordPress Desktop CI Failure for job "wp-desktop-source".

@monsieur-z please inspect this job's build steps for breaking changes at [this link](https://circleci.com/gh/Automattic/wp-calypso/813312). For temporal failures, you may try to "Rerun Workflow from Failed".

Please also ensure this branch is rebased off latest Calypso.

File: client/my-sites/checkout/checkout-system-decider.js [hunk 4]
```diff
@@ -43,11 +49,30 @@ export default function CheckoutSystemDecider( {
 	cart,
 	isWhiteGloveOffer,
 } ) {
-	const isJetpack = useSelector( ( state ) => isJetpackSite( state, selectedSite?.ID ) );
-	const isAtomic = useSelector( ( state ) => isSiteAutomatedTransfer( state, selectedSite?.ID ) );
+	const siteId = selectedSite?.ID;
+	const jetpackPlan = getPlanByPathSlug( product, GROUP_JETPACK );
```
Comment (monsieur-z) [inline client/my-sites/checkout/checkout-system-decider.js:53]: `product` is a string.

**Comment by robertf4** [comment]
I'm having trouble getting the message to display in one case. It seems to be a problem with `getPlanByPathSlug()`. 

To reproduce:
1. Buy Jetpack Personal on a new site.
2. Visit `/checkout/<site_slug>/jetpack_backup_daily`
3. `getPlanByPathSlug()` returns `undefined`, so the notice doesn't display.

**Review by robertf4 — COMMENTED** [review]
I'm having trouble getting the message to display in one case. It seems to be a problem with getPlanByPathSlug().

To reproduce:

1. Buy Jetpack Personal on a new site.
2. Visit /checkout/<site_slug>/jetpack_backup_daily
3. getPlanByPathSlug() returns undefined, so the notice doesn't display.

**Comment by monsieur-z** [comment]
@robertf4, from my understanding the scope of the issue was to display the notice when you have a Backup product (either _Daily_ or _Real-Time_) and are about to purchase a plan that already includes it.

In your example, it's the other way around: you have a plan and want to buy a Backup product. The UI of Calypso already prevents you from reaching the checkout page with this state, though it could happen from Jetpack.com indeed.

I can handle this in a different PR, if we feel it should be part of the fix too.

File: client/my-sites/checkout/checkout/secure-payment-form.jsx (hunk 3 above)
Comment (rcanepa) [reply client/my-sites/checkout/checkout/secure-payment-form.jsx:314]: I noticed the same thing.

**Comment by robertf4** [comment]
@monsieur-z I think it's fine to do in a separate PR but I think we do need to do it. The order the conflicting plans are purchased in isn't as important as as the fact that they are purchased simultaneously, and I know we have users who have purchased them in both orders.

I was looking at this Friday but didn't get very far, the plans-list doesn't account for products and so it's hard to know where to check for the product slug.

**Review by rcanepa — APPROVED** [review]
LGTM. I test this with several payment types and it worked as expected. 

I have one question though, why did you decide to check for a duplicate Backup feature inside the `CheckoutSystemDecider` component? I ask because I saw we are passing down the `infoMessage`  prop through three components that don't need it. The chain looks like `CheckoutSystemDecider > CheckoutContainer > Checkout > SecurePaymentForm`, and `SecurePaymentForm` is the component that actually makes use of the prop.

**Comment by rcanepa** [comment]
> @monsieur-z I think it's fine to do in a separate PR but I think we do need to do it. The order the conflicting plans are purchased in isn't as important as as the fact that they are purchased simultaneously, and I know we have users who have purchased them in both orders.
> 
> I was looking at this Friday but didn't get very far, the plans-list doesn't account for products and so it's hard to know where to check for the product slug.

I agree with this. Maybe this PR is not the place to fix the issue, but we should address it because users can easily get to the checkout page with a Backup product in their cart because they followed a link outside of Calypso.

**Comment by monsieur-z** [comment]
> I have one question though, why did you decide to check for a duplicate Backup feature inside the `CheckoutSystemDecider` component? I ask because I saw we are passing down the `infoMessage` prop through three components that don't need it. The chain looks like `CheckoutSystemDecider > CheckoutContainer > Checkout > SecurePaymentForm`, and `SecurePaymentForm` is the component that actually makes use of the prop.

Good point @rcanepa. The very same notice is used for the composite checkout (which you can see in this separate PR #44237), so it made sense to have the logic where the 2 checkout components were created.

**Comment by monsieur-z** [comment]
> I think it's fine to do in a separate PR but I think we do need to do it.

I agree @robertf4 and @rcanepa. We focused on releasing a solution for the main scenario first. I believe  I'll have time to handle this case this week too.

**Review by wp-desktop — DISMISSED** [review]
WordPress Desktop CI Failure for job "wp-desktop-source".

@monsieur-z please inspect this job's build steps for breaking changes at [this link](https://circleci.com/gh/Automattic/wp-calypso/820084). For temporal failures, you may try to "Rerun Workflow from Failed".

Please also ensure this branch is rebased off latest Calypso.

**Review by wp-desktop — CHANGES_REQUESTED** [review]
WordPress Desktop CI Failure for job "wp-desktop-mac".

@monsieur-z please inspect this job's build steps for breaking changes at [this link](https://circleci.com/gh/Automattic/wp-calypso/820228). For temporal failures, you may try to "Rerun Workflow from Failed".

Please also ensure this branch is rebased off latest Calypso.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/4120774

Thank you @monsieur-z for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/4120774).

**Comment by juzer28** [comment]
G

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/4120774).

**Pull request merged**